library/src/main/java/com/ricohapi/mstorage/MediaStorage.java

```diff
@@ -109,25 +110,41 @@ public void download(String mediaId, CompletionHandler<MediaContent> handler) {
 
     }
 
-//    public void list(Map<String, Objects> filter, CompletionHandler<MediaList> handler) {
-//        // TODO
-//    }
-
-    public void list(Map<String, String> params, CompletionHandler<MediaList> handler) {
+    public void list(Map<String, ?> params, CompletionHandler<MediaList> handler) {
         try {
             if (accessToken == null) {
                 throw new RicohAPIException(0, "wrong usage: use the connect method to get an access token.");
             }
             RicohAPIRequest request;
+            Map<String, String> header = new HashMap<>();
+            header.put("Authorization", "Bearer " + accessToken);
             if (params == null || params.isEmpty()) {
+                // GET /media
                 request = new RicohAPIRequest(ENDPOINT);
+                request.get(header);
             } else {
-                request = new RicohAPIRequest(ENDPOINT, params);
+                if (!params.containsKey(LIST_PARAM_KEY_FILTER) || params.get(LIST_PARAM_KEY_FILTER) == null) {
+                    // GET /media
+                    request = new RicohAPIRequest(ENDPOINT, params);
```

Comment:
No, we don't.
In API reference, those parameter is not listed in parameters:
* http://doc.mss.pflabo.info/api/1.2-devauth-preview2/index.html#!/Media/get_media
* https://github.com/ricohapi/media-storage-rest/blob/master/media.md#get-media

So this is right usage.